api/v1alpha2/virtualmachine_types.go

```diff
@@ -189,26 +189,23 @@ type VirtualMachineSpec struct {
 	//
 	// This field may be used to specify the name of a VirtualMachineImage
 	// or ClusterVirtualMachineImage resource. The resolver first checks to see
-	// if there is a ClusterVirtualMachineImage with the specified name. If no
+	// if there is a VirtualMachineImage with the specified name. If no
 	// such resource exists, the resolver then checks to see if there is a
-	// VirtualMachineImage resource with the specified name in the same
+	// ClusterVirtualMachineImage resource with the specified name in the same
 	// Namespace as the VM being deployed.
 	//
-	// This field is optional in the cases where there exists a sensible
-	// default value, such as when there is a single VirtualMachineImage
-	// resource available in the same Namespace as the VM being deployed.
-	//
-	// +optional
+	// This field may also be used to specify the display name (vSphere name) of
+	// a VirtualMachineImage or ClusterVirtualMachineImage resource. If the
+	// display name unambiguously resolves to a distinct VM image (among all
+	// existing VirtualMachineImages in the VM's namespace and all existing
+	// ClusterVirtualMachineImages), then a mutation webhook updates this field
+	// with the VM image resource name. If the display name resolves to multiple
+	// or no VM images, then the mutation webhook denies the request and outputs
+	// an error message accordingly.
 	ImageName string `json:"imageName,omitempty"`
 
 	// ClassName describes the name of the VirtualMachineClass resource used to
 	// deploy this VM.
-	//
-	// This field is optional in the cases where there exists a sensible
-	// default value, such as when there is a single VirtualMachineClass
-	// resource available in the same Namespace as the VM being deployed.
-	//
-	// +optional
```

Comment:
Please leave `// +optional` for now since it will be in the future. We already enforce this in the webhook. FWIW, you should also always keep `// +optional` and `omitempty` in sync.